settings.go

```diff
@@ -44,19 +60,46 @@ func newSettings(payload []byte, paths ...string) (Settings, error) {
 
 	data := gjson.GetManyBytes(payload, paths...)
 
-	deniedNames := mapset.NewThreadUnsafeSet()
+	var err error
+	allowedHostPaths := make([]HostPath, 0)
+
+	if data[0].String() == "" {
+		// empty settings
+		return Settings{
+			AllowedHostPaths: allowedHostPaths,
+		}, nil
+	}
+
 	data[0].ForEach(func(_, entry gjson.Result) bool {
-		deniedNames.Add(entry.String())
-		return true
+		dataHostPath := gjson.GetManyBytes([]byte(entry.String()),
+			"pathPrefix",
+			"readOnly",
+		)
+		if !dataHostPath[0].Exists() {
+			err = fmt.Errorf("pathPrefix key is missing")
```

Comment:
It could be done. Once it checks for  `initContainers` and all containers, it will be slower. While nice, I would just skip it (and other policies also bail on first error).

Reply:
Oh, here I was meaning the settings, but yes, it could be made extensive to the errors from the policy itself, but yeah... If you think is a refinement that can be done later is perfectly fine.

Reply:
Oh, I misunderstood. Done for settings in https://github.com/kubewarden/psp-hostpaths-policy/pull/1/commits/a0ce98e9286e0558250779c68fc44b8c42795aa1.